WeatherApp/WeatherHandler.swift

```diff
@@ -34,7 +34,80 @@ class WeatherHandler {
             }
             return
         }
-        print(cityURL)
+        
+        let request = URLRequest(url: cityURL)
+        let task = URLSession.shared.dataTask(with: request) {
+            (data, response, error) in
+            
+            if let jsonData = data {
+                do {
+                    guard
+                        let jsonDictionary = try JSONSerialization.jsonObject(with: jsonData, options: []) as? [String:AnyObject]
+                        else {
+                            DispatchQueue.main.async {
+                                completion(false)
+                            }
+                        return
+                    }
+                    
+                    let date = jsonDictionary["dt"]
+                    let weatherDate = Date(timeIntervalSince1970: date!.doubleValue)
+                    let weatherArray = jsonDictionary["weather"] as? [[String: AnyObject]]
+                    let weatherDictionary = weatherArray?.first
+                    let weatherCondition = weatherDictionary!["main"] as!  String
+                    let conditionID = weatherDictionary!["id"] as! Int
+                    let weatherMain = jsonDictionary["main"] as! [String: Double]
+                    let tempValue = weatherMain["temp"] ?? 0
+                    let minValue = weatherMain["temp_min"] ?? 0
+                    let maxValue = weatherMain["temp_max"] ?? 0
+                    let currTemperature = Measurement(value: tempValue, unit: UnitTemperature.celsius)
+                    let minTemperature = Measurement(value: minValue, unit: UnitTemperature.celsius)
+                    let maxTemperature = Measurement(value: maxValue, unit: UnitTemperature.celsius)
+                    
+                    let currentWeatherData = WeatherData(date: weatherDate, avgTemp: currTemperature, minTemp: minTemperature, maxTemp: maxTemperature, condition: weatherCondition, conditionID: conditionID)
+                    
+                    city.weatherData.append(currentWeatherData)
+                    
+                    DispatchQueue.main.async {
+                        completion(true)
+                    }
+                    
+                }
+                catch let error {
+                    DispatchQueue.main.async {
+                        completion(false)
+                    }
+                    print(error)
+                }
+            } else {
+                DispatchQueue.main.async {
+                    completion(false)
+                }
+            }
+        }
+        task.resume()
+    }
+    
+    func weatherForCity(city: City) {
```

Comment:
this should be named: updateWeather(forCity city: City)

weatherForCity implies that the function returns the weather.